openreview/venue/invitation.py

```diff
@@ -1818,6 +1818,144 @@ def set_submission_revision_invitation(self, submission_revision_stage=None):
         self.save_invitation(invitation, replacement=True)
         return invitation
 
+    def set_custom_stage_invitation(self):
+
+        venue_id = self.venue_id
+        custom_stage = self.venue.custom_stage
+        custom_stage_invitation_id = self.venue.get_invitation_id(custom_stage.name)
+        custom_stage_cdate = tools.datetime_millis(custom_stage.start_date if custom_stage.start_date else datetime.datetime.utcnow())
+        custom_stage_duedate = tools.datetime_millis(custom_stage.due_date) if custom_stage.due_date else None
+        custom_stage_expdate = tools.datetime_millis(custom_stage.exp_date) if custom_stage.exp_date else None
+        if not custom_stage_expdate:
+            custom_stage_expdate = tools.datetime_millis(custom_stage.due_date + datetime.timedelta(minutes = SHORT_BUFFER_MIN)) if custom_stage.due_date else None
+
+        content = custom_stage.content
+
+        custom_stage_replyto = custom_stage.get_reply_to()
+        custom_stage_source = custom_stage.get_source_submissions()
+
+        if custom_stage_replyto == 'forum':
+            paper_invitation_id = self.venue.get_invitation_id(name=custom_stage.name, number='${2/content/noteNumber/value}')
+            with_invitation = self.venue.get_invitation_id(name=custom_stage.name, number='${6/content/noteNumber/value}')
+            reply_to = '${4/content/noteId/value}'
+        else:
+            paper_invitation_id = self.venue.get_invitation_id(name=custom_stage.name, prefix='${2/content/replytoSignatures/value}')
+            with_invitation = self.venue.get_invitation_id(name=custom_stage.name, prefix='${6/content/replytoSignatures/value}')
+            reply_to = '${4/content/replyto/value}' 
```

Comment:
so you are going to create one invitation per reply instead one invitation per forum?

what happens if there are reviews created after that?

Reply:
Yes, I just remembered we spoke about this; this would be very heplful for SAC agreement. I will make this change. What will we do about this showing as completed in tasks when there are more tasks to be completed? For example, review rating?